src/revlanguage/main.cpp

```diff
@@ -28,14 +28,38 @@ using po::variables_map;
 
 std::string usage()
 {
-    return "Usage: rb [OPTIONS]\n       rb [OPTIONS] <file1> [<file2> ...]\n";
-    // Other usages not mentioned
+    std::string usage =
+        "Usage: rb [OPTIONS]\n"
+        "       rb [OPTIONS] <file1> [<file2> ...] [--args <arg1> [<arg2> ...]]";
+
+    return usage;
+}
+
+
+std::string usage_examples()
+{
+    std::string usage_examples =
+        "Usage examples:\n"
+        "   1. rb\n"
+        "   2. rb --args 1 2\n"
+        "   3. rb script.Rev\n"
+        "   4. rb --file script.Rev                         # equivalent to 3\n"
+        "   5. rb --cmd script.Rev                          # equivalent to 3 and 4 but runs in batch mode\n"
+        "   6. rb script.Rev --args 1 2\n"
+        "   7. rb --args 1 2 --file script.Rev              # equivalent to 6\n"
```

Comment:
For `7. rb --args 1 2 --file script.Rev`, users should actually write `rb script.Rev --args 1 2`.  So I think we should drop this as well.

Reply:
I'd like to keep this one, since it was one of the cases mentioned in #360:

> It took me quite some time to work out that `--args` simply fills a vector called `args` [...] and that if you do that then `rb [OPTIONS] <file1> [<file2> ...]` does not actually work (because everything after `--args` is used to fill `args`) and you need to specify `--file` instead.

Reply:
It seems like the issue here is that
- the Usage string said to put "OPTIONS" before files.
- however, if you write actually do that by writing `rb --args arg1 file`, then this actually does not do what was promised, because `--args` consumes all arguments after it until it hits another words starting with `-`.
- as a result, the user had to try a whole bunch of random things until figuring out something that worked.

The solution they found was to insert `--file` here, yielding something like your `7. rb --args 1 2 --file script.Rev`.  But of course an equally good solution would be to write `rb file --args arg1`, which contradicts the previous Usage string.  So it doesn't seem to me that _specific_ they found is required, so much that:
- it needs to be obvious how to specify a file and an argument
- it needs to be obvious why `rb --args arg1 file` is not the right way to do that.
- the Usage string needs to not imply that you should do things in a way that doesn't work.

That said, although the solution they found doesn't seem to me to be required by #360, it obviously isn't ruled out by it either.

Reply:
I agree it's not required – it's just a valid use case that someone has previously found useful. For me, the fact the two solutions work equally well is a good enough reason to include them both.

I guess it comes down to whether we want Usage examples to... well, list valid usage examples, or _educate_ users on how to do things _properly_. I think the latter is a job for a tutorial rather than the help page.

Reply:
There are certainly a lot of valid usage examples that are not listed.  We don't list `-b` or `-j` or `--setOption`, and we don't list things like `rb file2 --file file3 --cmd file1 arg --file file4 --file 5`.  I'd be in favor of adding examples for `-b`, `-j`, and `--setOption`, but less in favor of listing `rb file2 --file file3 --cmd file1 arg --file file4 --file 5`.

However, we do have different preferences in that I prefer the usage examples to be more information-dense.

@bjoelle Do you have an opinion here?

Reply:
in general I think it's fine to propose to users one "canonical" way to do things and not list all potential alternatives
on the other hand, I would be in favour of keeping 7. because putting the input file at the very end is a very common Linux/command-line way of doing things so it will likely be the syntax that will come naturally to a lot of people - the only exception I can think of is Rscript, but that one doesn't need a special `--args` flag either

Reply:
actually if we're thinking about reorganizing the input options at some points Rscript may not be the worst model

Reply:
I do hope we can reorganize the input options, and I like what Rscript does.

Since the changes won't be backwards-compatible, we will have to check how many people rely on doing things like `rb script1 script2`.  With the Rscript approach, that would have to be changed to one of:
- put `source(script2)` at the end of `script1`, `rb  script1`
- `cat script1 script2 | rb`
- `rb -e 'source(script1)' -e 'source(script2)'`

Reply:
Of course, reorganizing the input options is a separate project.

Reply:
`rb script1 script2`

I had never even heard of doing this.

Reply:
Hence this PR :)